.github/workflows/common.yml

```diff
@@ -0,0 +1,125 @@
+          . ${{ env.SPACK_PATH }}/share/spack/setup-env.sh
+          spack config --scope site add config:ccache:true
+          spack config --scope site add config:install_tree:padded_length:128
+          spack config --scope site add packages:all:require:'target=x86_64_v3'
+          cat <<EOF > ${{ env.SPACK_PATH }}/etc/spack/mirrors.yaml
+          mirrors:
+            ghcr-buildcache:
+              url: oci://ghcr.io/ghex-org/spack-${{ matrix.spack-version }}-buildcache-${{ env.OS_NAME }}-${{ env.OS_VERSION }}
+              signed: false
+              access_pair:
+                id_variable: GITHUB_USER
+                secret_variable: GITHUB_TOKEN
+          EOF
+
+          spack compiler find
+          spack repo add ${{ env.REPO_PATH }}
+          spack reindex
+          spack buildcache list
+
+      - name: Build dependencies
+        run: |
+          . ${{ env.SPACK_PATH }}/share/spack/setup-env.sh
+          echo "spack spec -I ${{ inputs.package-name }}@${{ inputs.package-version }} ${{ inputs.package-variants }}"
+          spack spec -I ${{ inputs.package-name }}@${{ inputs.package-version }} ${{ inputs.package-variants }}
+          spack install --only dependencies ${{ inputs.package-name }}@${{ inputs.package-version }} ${{ inputs.package-variants }}
+
+      - name: Push packages to buildcache
+        env:
+          GITHUB_USER: ${{ github.actor }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          . ${{ env.SPACK_PATH }}/share/spack/setup-env.sh
+          spack --color never find --format '{/hash};' | sed 's/;[[:space:]]*//g' | xargs -I{} spack buildcache push --base-image ${{ env.OS_NAME }}:${{ env.OS_VERSION }} --only package ghcr-buildcache  {}
+        if: ${{ !cancelled() }}
+        continue-on-error: true
+
+      - name: Build package without testing
+        if: ${{ !inputs.test-package }}
+        run: |
+          . ${{ env.SPACK_PATH }}/share/spack/setup-env.sh
+          spack install --verbose --show-log-on-error ${{ inputs.package-name }}@${{ inputs.package-version }} ${{ inputs.package-variants }}
+
+      - name: Build and test package
+        if: ${{ inputs.test-package }}
+        run: |
+          . ${{ env.SPACK_PATH }}/share/spack/setup-env.sh
+          spack install --test=root --verbose --show-log-on-error ${{ inputs.package-name }}@${{ inputs.package-version }} ${{ inputs.package-variants }}
+
+      #- name: Push packages and update index
+      #  env:
+      #    GITHUB_USER: ${{ github.actor }}
+      #    GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      #  run: |
+      #    . ${{ env.SPACK_PATH }}/share/spack/setup-env.sh
+      #    spack buildcache push --base-image ${{ env.OS_NAME }}:${{ env.OS_VERSION }} --update-index --only dependencies ghcr-buildcache ${{ inputs.package-name }}@${{ inputs.package-version }} ${{ inputs.package-variants }}
+      #  if: ${{ !cancelled() }}
+      #  continue-on-error: true
```

Comment:
Is this necessary to keep or can it be removed?

Reply:
will remove